src/Components/Facility/AddInventoryForm.tsx

```diff
@@ -105,6 +135,46 @@ export const AddInventoryForm = (props: any) => {
     }
   }, [state.form.id]);
 
+  const defaultUnitConverter = (unitData: any) => {
+    const unitName = data[Number(unitData.item - 1)].allowed_units?.filter(
+      (u: any) => Number(u.id) === Number(unitData.unit)
+    )[0].name;
+    if (unitName === "Dozen") {
+      return Number(unitData.quantity) * 12;
+    }
+    if (unitName === "Gram") {
+      return Number(unitData.quantity) / 1000;
+    }
+    return Number(unitData.quantity);
+  };
+
+  const stockValidation = (data: any) => {
+    if (inventory && inventory.length) {
+      const stockBefore = inventory.filter(
+        (inventoryItem: any) =>
+          Number(inventoryItem.item_object.id) === Number(data.item)
+      );
+      if (stockBefore.length === 0) {
+        setStockError("No Stock Available ! Please Add Stock.");
+        setIsLoading(false);
+        return false;
+      } else {
+        const stockAfterQuantity = defaultUnitConverter(data);
+        if (stockAfterQuantity > Number(stockBefore[0].quantity)) {
+          setStockError("Stock Insufficient ! Please Add Stock.");
+          setIsLoading(false);
+          return false;
+        }
+        setStockError("");
+        return true;
+      }
+    } else if (inventory && inventory.length === 0) {
+      setStockError("No Stock Available !");
+      setIsLoading(false);
+      return false;
+    }
+  };
```

Comment:
can this function be refactored? now it is kinda hard to follow (less readable)

for starters, remove else when return is used inside if, 

feel free to go your way